service/api_task.go

```diff
@@ -187,6 +187,20 @@ func (as *APIServer) EndTask(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// For a single-host task group, if a task fails, block and dequeue later tasks in that group.
```

Comment:
Could you add a comment here explaining that this must be called before `model.MarkEnd`? Someone might move things around in this function and not realize that order matters, since the bug you found is pretty subtle.

Reply:
done.